public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.ts

```diff
@@ -32,7 +32,7 @@ export class VariableSupport extends CustomVariableSupport<DataSource, AzureMoni
         try {
           const templateVariablesResults = await this.callGrafanaTemplateVariableFn(queryObj.grafanaTemplateVariableFn);
           return {
-            data: templateVariablesResults ? [toDataFrame(templateVariablesResults)] : [],
+            data: (templateVariablesResults && templateVariablesResults.length > 0) ? [toDataFrame(templateVariablesResults)] : [],
```

Comment:
nit:

```suggestion
            data: templateVariablesResults?.length ? [toDataFrame(templateVariablesResults)] : [],
```

Reply:
Thanks for the suggestions, I have added the test and incorporated this